environment-api/src/main/java/com/sequenceiq/environment/api/v1/environment/model/request/azure/AzureResourceEncryptionParameters.java

```diff
@@ -0,0 +1,74 @@
+package com.sequenceiq.environment.api.v1.environment.model.request.azure;
+
+import javax.validation.constraints.Pattern;
+
+import com.sequenceiq.environment.api.doc.environment.EnvironmentModelDescription;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel(value = "AzureResourceEncryptionV1Parameters")
+public class AzureResourceEncryptionParameters {
+
+    @ApiModelProperty(EnvironmentModelDescription.KEY_VAULT_ID)
+    @Pattern(regexp = "(^https:\\/\\/[a-zA-Z-][0-9a-zA-Z-]*\\.vault\\.azure\\.net)",
+            message = "Key Vault Id should be of format 'https://<key-vault-name>.vault.azure.net/'." +
+                    "Vault name must only contain alphanumeric characters and dashes and cannot start with a number.")
+    private String keyVaultId;
```

Comment:
Is it really an URL needed here? The associated [REST API](https://docs.microsoft.com/en-us/rest/api/compute/diskencryptionsets/createorupdate#sourcevault) says this should be an ARM Resource ID `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults/{vaultName}`.

Reply:
Agree, It should be vaultName only.
Also, keyName should be a URL of format `https://<vaultName>.vault.azure.net/keys/juhig/d0c42094a2ef4ad99c2ccbb6663b949d` , updating both.

Reply:
No, I do not think `{vaultName}` is enough. You cannot guess `{subscriptionId}` and `{resourceGroupName}` that are needed for the vault Resource ID.

Reply:
`All resources related to customer-managed keys (Azure Key Vaults, disk encryption sets, VMs, disks, and snapshots) must be in the same subscription and region.`
We can frame the Resource ID, as the `subscriptionId` and `resourceGroupName` would be same for all resources.

Reply:
Ah okay, then it makes sense.